pkg/scheduler/scheduler.go

```diff
@@ -79,81 +70,175 @@ func (sched *Scheduler) Cache() schedulercache.Cache {
+		o.disablePreemption = disablePreemption
+	}
+}
+
+// WithPercentageOfNodesToScore sets percentageOfNodesToScore for Scheduler, the default value is 50
+func WithPercentageOfNodesToScore(percentageOfNodesToScore int32) Option {
+	return func(o *schedulerOptions) {
+		o.percentageOfNodesToScore = percentageOfNodesToScore
+	}
+}
+
+// WithBindTimeoutSeconds sets bindTimeoutSeconds for Scheduler, the default value is 100
+func WithBindTimeoutSeconds(bindTimeoutSeconds int64) Option {
+	return func(o *schedulerOptions) {
+		o.bindTimeoutSeconds = bindTimeoutSeconds
+	}
+}
+
+var defaultSchedulerOptions = schedulerOptions{
+	schedulerName:                  v1.DefaultSchedulerName,
+	hardPodAffinitySymmetricWeight: v1.DefaultHardPodAffinitySymmetricWeight,
+	enableEquivalenceClassCache:    false,
+	disablePreemption:              false,
+	percentageOfNodesToScore:       schedulerapi.DefaultPercentageOfNodesToScore,
+	bindTimeoutSeconds:             BindTimeoutSeconds,
 }
 
-// Config is an implementation of the Scheduler's configured input data.
-// TODO over time we should make this struct a hidden implementation detail of the scheduler.
-type Config struct {
-	// It is expected that changes made via SchedulerCache will be observed
-	// by NodeLister and Algorithm.
-	SchedulerCache schedulercache.Cache
-	// Ecache is used for optimistically invalid affected cache items after
-	// successfully binding a pod
-	Ecache     *equivalence.Cache
-	NodeLister algorithm.NodeLister
-	Algorithm  algorithm.ScheduleAlgorithm
-	GetBinder  func(pod *v1.Pod) Binder
-	// PodConditionUpdater is used only in case of scheduling errors. If we succeed
-	// with scheduling, PodScheduled condition will be updated in apiserver in /bind
-	// handler so that binding and setting PodCondition it is atomic.
-	PodConditionUpdater PodConditionUpdater
-	// PodPreemptor is used to evict pods and update pod annotations.
-	PodPreemptor PodPreemptor
-
-	// NextPod should be a function that blocks until the next pod
-	// is available. We don't use a channel for this, because scheduling
-	// a pod may take some amount of time and we don't want pods to get
-	// stale while they sit in a channel.
-	NextPod func() *v1.Pod
-
-	// WaitForCacheSync waits for scheduler cache to populate.
-	// It returns true if it was successful, false if the controller should shutdown.
-	WaitForCacheSync func() bool
-
-	// Error is called if there is an error. It is passed the pod in
-	// question, and the error
-	Error func(*v1.Pod, error)
-
-	// Recorder is the EventRecorder to use
-	Recorder record.EventRecorder
-
-	// Close this to shut down the scheduler.
-	StopEverything chan struct{}
-
-	// VolumeBinder handles PVC/PV binding for the pod.
-	VolumeBinder *volumebinder.VolumeBinder
-
-	// Disable pod preemption or not.
-	DisablePreemption bool
+// New returns a Scheduler
+// TODO：Once we have the nice constructor，we should modify cmd/kube-scheduler to use it.
+func New(client clientset.Interface,
```

Comment:
We will eventually need to rework the unit tests in this package to cover this constructor instead of the other ones. It would be nice to see some test coverage in this PR. At minimum it would be nice to cover `CreateFromProvider`

Reply:
I have add test for this constructor. Maybe you can review it and Is it what you want? please comment your advice~ thank you